ci-operator/step-registry/interop/mtr/execute/README.md

```diff
@@ -0,0 +1,164 @@
+
+## Process
+
+This script can be separated into 5 sections - Define required variables, create configuration file, install MTR tests, start the local FTP server, execute tests, and stop the local FTP server.
+
+### Define Required Variables
+
+The following code snippet is used to define the variables needed to [create the MTR test configuration file](#create-mtr-test-configuration-file). The variables defined in this step come from files in the `SHARED_DIR` and [credentials](#credentials) from Vault.
+
+```bash
+SECRETS_DIR="/tmp/secrets"
+APPS_URL=$(cat ${SHARED_DIR}/apps_url)
+MTR_USERNAME=$(cat ${SECRETS_DIR}/mtr/mtr-username)
+MTR_PASSWORD=$(cat ${SECRETS_DIR}/mtr/mtr-password)
+FTP_USERNAME=$(cat ${SECRETS_DIR}/ftp/ftp-username)
+FTP_PASSWORD=$(cat ${SECRETS_DIR}/ftp/ftp-password)
+FTP_HOST=$(cat ${SECRETS_DIR}/ftp/ftp-host)
+APP_HOSTNAME="http://${APPS_URL}"
+OCP_HOSTNAME="http://mtr-mtr.${APPS_URL}/"
+OCP_SECURE_HOSTNAME="https://secure-mtr-mtr.${APPS_URL}/"
+SELENIUM_EXECUTOR=$(cat ${SHARED_DIR}/selenium-executor)
+```
+
+### Create MTR Test Configuration File
+
+The following code snippet is used to create the MTR test configuration file needed to execute the tests properly against the test cluster. The file is created by replacing values in a [pre-defined yaml file within the container](https://github.com/windup/windup_integration_test/blob/mtr/dockerfiles/interop/env.yaml) using the `sed` command along with [variables](#define-required-variables) defined earlier in the script.
+
+```bash
+echo "Replacing values in config file"
+sed -i "s#REPLACE_HOSTNAME#${APP_HOSTNAME}#" $CONFIG_FILE
+sed -i "s#REPLACE_OCP_HOSTNAME#${OCP_HOSTNAME}#" $CONFIG_FILE
+sed -i "s#REPLACE_OCP_SECURE_HOSTNAME#${OCP_SECURE_HOSTNAME}#" $CONFIG_FILE
+sed -i "s#REPLACE_FTP_HOST#${FTP_HOST}#" $CONFIG_FILE
+sed -i "s#REPLACE_FTP_USERNAME#${FTP_USERNAME}#" $CONFIG_FILE
+sed -i "s#REPLACE_FTP_PASSWORD#${FTP_PASSWORD}#" $CONFIG_FILE
+sed -i "s#REPLACE_EXECUTOR#${SELENIUM_EXECUTOR}#" $CONFIG_FILE
+sed -i "s#REPLACE_MTR_PASSWORD#${MTR_PASSWORD}#" $CONFIG_FILE
+sed -i "s#REPLACE_MTR_USER#${MTR_USERNAME}#" $CONFIG_FILE
+sed -i "s#REPLACE_MTR_VERSION#${MTR_VERSION}#" $CONFIG_FILE
+```
+
+### Install MTR Tests
+
+The following code snippet uses `pip` to install the MTR tests from the `/tmp/integration_tests` directory within the container. These tests come from the the [windup/windup_integration_test](https://github.com/windup/windup_integration_test.git) repository maintained by MTR product QE. These tests must be installed in this script rather than when the container image is built because OpenShift runs these containers using a user that ends up not having access to [modify the configuration file](#create-mtr-test-configuration-file) needed to execute these tests. Because that configuration file changes with every run, we have to modify it *then* install the tests.
```

Comment:
`These tests must be installed in this script rather than when the container image is built because OpenShift runs these containers using a user that ends up not having access to ....` - can't we have a service account with the appropriate credentials? otherwise we will end up implementing in custom scripts things that should be pre-configured in the image

Reply:
It isn't a service account issue, it is an issue with the user that OpenShift runs the container with. I cannot modify the configuration file in this script if I run pip in the Dockerfile.